include/spi.h

```diff
@@ -152,25 +183,32 @@ struct spi_cs_control {
  *     mode                [ 1 : 3 ]   - Polarity, phase and loop mode.
  *     transfer            [ 4 ]       - LSB or MSB first.
  *     word_size           [ 5 : 10 ]  - Size of a data frame in bits.
- *     lines               [ 11 : 12 ] - MISO lines: Single/Dual/Quad.
+ *     lines               [ 11 : 12 ] - MISO lines: Single/Dual/Quad/Octal.
  *     cs_hold             [ 13 ]      - Hold on the CS line if possible.
  *     lock_on             [ 14 ]      - Keep resource locked for the caller.
- *     eeprom              [ 15 ]      - EEPROM mode.
+ *     cs_active_high      [ 15 ]      - Active high CS logice.
  * @param slave is the slave number from 0 to host controller slave limit.
  * @param cs is a valid pointer on a struct spi_cs_control is CS line is
  *    emulated through a gpio line, or NULL otherwise.
  *
- * @note cs_hold, lock_on and eeprom_rx can be changed between consecutive
+ * @note cs_hold and lock_on can be changed between consecutive
  * transceive call.
  */
 struct spi_config {
 	struct device	*dev;
 
+#ifdef CONFIG_SPI_SLAVE
+	union {
+		u32_t		frequency;
+		spi_slave_cb	slave_cb;
+	};
+#else
 	u32_t		frequency;
+#endif /* CONFIG_SPI_SLAVE */
 	u16_t		operation;
```

Comment:
It seems like we need to be able to support SPI slave & master at the same time.  I could see a device w/multiple SPI controllers with one connected to sensors(master) and another to a host processor (slave).

Reply:
Some controllers do support both at same time, some only one of the 2 modes. 
That's a driver issue. It's already taken care of in spi_context (in a patch which is not here actually, I'll update that). It really does not need to be exposed in the user API itself.

Reply:
Actually, the patch was missing 4 important lines, that got lost in one rebase: the enum that dictates runtime mode support (master/slave). Up to the device drivers then to use it (dw driver will be updated along with slave support). As it is not part of the public API, it could be changed/improved.

Reply:
Sounds good, just don't want us to preclude the use model I suggested.